### PR TITLE
add Athena URL parameter to configure AWS region

### PR DESCRIPTION
**Pull request by bpdohall**
Contributes to gravitational/cloud#4658

RFD: [Multi-region DynamoDB - Teleport Core](https://github.com/gravitational/cloud/blob/3c066aa2bd3f765a38dff96942500a619e525828/rfd/xxxx-Multi-Region-Dynamo.md#teleport-core)

We're adding support for choosing the region of DynamoDB tables and Auth service in Teleport Cloud. This PR provides the ability to override the yaml configuration `teleport.storage.region` by including a new URL parameter in the 
Athena URI (`teleport.storage.audit_events_uri`).

Athena audit logs were introduced in v13, so backport to v12 is not required.

Signed-off-by: Brendan O'Halloran <brendan.ohalloran@goteleport.com>

### CODE REVIEW

**Review by tobiaszheller — APPROVED** [review]

**Comment by tobiaszheller** [comment]
@bpdohall does it mean that sqs/sns/glue will be created also in other region?
https://github.com/gravitational/teleport/blob/409e08cc0f56e6f7a006718ae164022878f00e66/lib/events/athena/athena.go#L251-L257
with your change will also update aws config region for athena part

**Comment by bpdohall** [comment]
> @bpdohall does it mean that sqs/sns/glue will be created also in other region?

Yes, that's correct - the intent is to have the region override affect all the Athena related resources. We don't have the infrastructure to support all the regions for all resources yet, so we need to be able to pin the region while the Dynamo tables transition to multi-region.

**Review by evanfreed — COMMENTED** [review]
@bpdohall would you mind adding a unit test?

**Review by evanfreed — APPROVED** [review]

**Review by strideynet — APPROVED** [review]
Approved contingent on `TestConfig_SetFromURL` including a modification to cover this case :)

File: lib/events/athena/athena.go [hunk 1]
```diff
@@ -276,6 +276,10 @@ func (cfg *Config) SetFromURL(url *url.URL) error {
 	}
 	cfg.Database, cfg.TableName = splitted[0], splitted[1]
 
+	region := url.Query().Get("region")
+	if region != "" {
+		cfg.Region = region
+	}
```
Comment (tigrato) [inline lib/events/athena/athena.go:282]: a simple nit: 
```suggestion
	if region := url.Query().Get("region");  region != "" {
		cfg.Region = region
	}
```

**Review by tigrato — APPROVED** [review]

**Pull request merged**